---
title: "DEG-tables_merge_filter"
author: "Veronica Radice"
date: "13/04/2021"
output: html_document
---

# Aim:  cross-species DEG comparison
## Comparison of potential shared response

### Import libraries
```{r}
library(tidyverse)
library(SummarizedExperiment)
library(cowplot)
```


### Set working directory
```{r}
setwd("~/Documents/Rprojects/postdoc\ Rprojects/ODU_postdoc_Rprojects/Paper_Ojo_gene-expression/Porites_astreoides/orthofinder/")
```

#### import Past seq2iso table
Carly Kenkel's P. astreoides transcriptome reference
```{r}
past_seq2iso <- read.table("~/Documents/Postdoc/ODU_postdoc/projects/Ojo_gene-expression/database/pastreoides_2014/pastreoides_may2014/past_seq2iso_suffixed.tab", header = FALSE, sep = "\t", stringsAsFactors=T)

past_seq2iso <- past_seq2iso %>% 
  dplyr::rename(isotig = V1,
         gene = V2)

head(past_seq2iso)
```

#### import Ppor seq2iso 
Same as Past table, but with _Ppor suffix
```{r}
ppor_seq2iso <- read.table("~/Documents/Postdoc/ODU_postdoc/projects/Ojo_gene-expression/database/pastreoides_2014/pastreoides_may2014/past_seq2iso_suffixed_Ppor.tab", header = FALSE, sep = "\t", stringsAsFactors=T)

ppor_seq2iso <- ppor_seq2iso %>% 
  dplyr::rename(isotig_Ppor = V1,
         gene_Ppor = V2)

head(ppor_seq2iso)
```

Same transcriptome reference, just used different suffixes for mapping of the different species
```{r}
seq2iso <- cbind(past_seq2iso, ppor_seq2iso)
head(seq2iso)
```

## Import transformed counts

```{r}
Sid.counts.transformed.vst <- readRDS(file = "~/Documents/Rprojects/postdoc\ Rprojects/ODU_postdoc_Rprojects/Paper_Ojo_gene-expression/Siderastrea/host/counts_vst.BlindFalse_Sid_Host.rds")
```


```{r}
Past.counts.transformed.vst <- readRDS(file = "~/Documents/Rprojects/postdoc\ Rprojects/ODU_postdoc_Rprojects/Paper_Ojo_gene-expression/Porites_astreoides/host/counts_vst.BlindFalse_Past_Host.rds")
```


```{r}
Ppor.counts.transformed.vst <- readRDS(file = "~/Documents/Rprojects/postdoc\ Rprojects/ODU_postdoc_Rprojects/Paper_Ojo_gene-expression/Porites_porites/host/counts_vst.BlindFalse_Ppor_Host.rds")
```


##############################################################################

# 1. Merge DESeq2 output with OrthoFinder output

Use Sid.v.Past_longestContig orthologue output.

## Import OrthoFinder orthogroups with species orthologues

Output file from OrthoFinder2:
    N0.tsv is a tab separated text file. Each row contains the genes belonging to a single orthogroup. The genes from each orthogroup are organized into columns, one per species. Additional columns give the HOG (Hierarchical Orthogroup) ID and the node in the gene tree from which the HOG was determined (note, this can be above the root of the clade containing the genes). This file effectively replaces the orthogroups in Orthogroups/Orthogroups.tsv from Markov clustering using MCL.
    
*Hierarchical orthogroup splitting:*
   When analysing the gene trees, a nested hierarchical group (any HOG other than N0, the HOG at the level of the last common ancestor of all species) may sometimes have lost its genes from the earliest diverging species and then duplicated before the first extant genes. The two first diverging clades will then be paralogous even though the evidence suggests they belong to the same HOG. For most analyses it is often better to split these clades into separate groups.
 
```{r}
orthogroups <- read.table("~/Documents/Rprojects/postdoc\ Rprojects/ODU_postdoc_Rprojects/Paper_Ojo_gene-expression/Porites_astreoides/orthofinder/Sid.v.Past_longestContig/Phylogenetic_Hierarchical_Orthogroups/N0.tsv", header = TRUE, sep = "\t", stringsAsFactors=F) #row.names = 1

orthogroups <- orthogroups %>% 
  dplyr::rename(orthogroup = HOG,
         gene_Sid = X19222_Sid_GoodCoral_500lnThresh_Final,
         isotig = X29422_past_LongestContig_suffixed)

dim(orthogroups)
```

```{r}
head(orthogroups)
```


select comparison gene name columns
```{r}
orthogroups <- orthogroups[, c(1,4:5)]
rownames(orthogroups) <- orthogroups$orthogroup
head(orthogroups)
```

```{r}
# Split the comma-separated paralogs
ortho.split <- strsplit(as.matrix(orthogroups),", ")
# it's now a list of character vectors

# Add dimensions:
dim(ortho.split) <- dim(orthogroups)

# Add row and column names:
dimnames(ortho.split) <- dimnames(orthogroups)
```

*List length: 13,305*


### To look at sequences for a specific orthogroup:

#### Example 1:1 orthogroup
```{r}
ortho.split["N0.HOG0000000", ]
```

#### Example multi-gene orthogroup
```{r}
ortho.split["N0.HOG0000021", ]
```

list of orthogroups 
```{r}
names_orthogroups <- row.names(apply(ortho.split,2,sapply,length))
```

### Check the sizes of the orthogroups

make a matrix containing the number of genes for each species in each orthogroup
```{r}
orthoSize <- apply(ortho.split,2,sapply,length)
orthoSize <- orthoSize[, 2:3]
```


check the number of in-paralogs in each species in orthogroup OG0000018:
```{r}
orthoSize["N0.HOG0000021", ]
```

plot how many orthogroups are represented in each transcriptome:
```{r}
plot(table(apply(orthoSize>0,1,sum)),ylab="no. orthogroups",xlab="no. transcriptomes")
```

**1:1 orthogroups**
how many orthogroups have exactly one ortholog in each species
```{r}
sum(apply(orthoSize==1,1,all)) # number of 1:1 groups
```

*If you are comparing more than 2 species*
calulate the number of orthogroups with exactly one ortholog in exactly two of the species
```{r}
#sum(apply(grpSize==1,1,sum)==2 & apply(grpSize > 1,1,sum)==0)
```


## A)  1:1 orthogroups

```{r}
names_1.to.1 <- as.data.frame(apply(orthoSize==1,1,all))

names_1.to.1 <- names_1.to.1 %>% 
  filter(names_1.to.1$`apply(orthoSize == 1, 1, all)` == TRUE)

names_1.to.1$orthogroups <- rownames(names_1.to.1)
names_1.to.1 <- names_1.to.1[ ,2]
```


```{r}
ortho_1.to.1  <- orthogroups[names_1.to.1, ]

ortho_1.to.1[sapply(ortho_1.to.1, is.character)] <- lapply(ortho_1.to.1[sapply(ortho_1.to.1, is.character)], as.factor)

dim(ortho_1.to.1)
```


```{r}
head(ortho_1.to.1)
```

```{r}
#write.csv(file = "Orthogroups_Sid-Past_1-1.csv", x = ortho_1.to.1)
```


1:1 Past
```{r}
ortho.1.to.1_Past.isotigs <- past_seq2iso %>% 
  inner_join(ortho_1.to.1, by = "isotig")

head(ortho.1.to.1_Past.isotigs)
```

```{r}
#write.csv(file = "Orthogroups_Sid-Past_1-1_Past-isotigs.csv", x = ortho.1.to.1_Past.isotigs)
```

1:1 Ppor
```{r}
names_1.to.1_ppor <- as.data.frame(apply(orthoSize==1,1,all))

names_1.to.1_ppor <- names_1.to.1_ppor %>%
  filter(names_1.to.1_ppor$`apply(orthoSize == 1, 1, all)` == TRUE)

names_1.to.1_ppor$orthogroups <- rownames(names_1.to.1_ppor)
names_1.to.1_ppor <- names_1.to.1_ppor[ ,2]

ortho_1.to.1_ppor  <- orthogroups[names_1.to.1_ppor, ]

ortho_1.to.1_ppor[sapply(ortho_1.to.1_ppor, is.character)] <- lapply(ortho_1.to.1_ppor[sapply(ortho_1.to.1_ppor, is.character)], as.factor)

ortho.1.to.1_Ppor.isotigs <- seq2iso %>% 
  inner_join(ortho_1.to.1_ppor, by = "isotig")

ortho.1.to.1_Ppor.isotigs <- ortho.1.to.1_Ppor.isotigs[ , 3:6]

ortho.1.to.1_Ppor.isotigs <- ortho.1.to.1_Ppor.isotigs %>% 
  dplyr::rename(isotig = isotig_Ppor,
                gene = gene_Ppor)

head(ortho.1.to.1_Ppor.isotigs)
```

```{r}
#write.csv(file = "Orthogroups_Sid-Past_1-1_Ppor-isotigs.csv", x = ortho.1.to.1_Ppor.isotigs)
```


## B)  Multi-orthogroup dataframe

```{r}
ortho.df <- as.data.frame(ortho.split)
```


#### filter out orthogroups that only belong to one species or the other
```{r}
ortho.df <- subset(ortho.df, ortho.df$gene_Sid != "character(0)")

ortho.df <- subset(ortho.df, ortho.df$isotig != "character(0)")

dim(ortho.df)
```

check for any 1:1 orthogroups 
```{r}
all(rownames(names_1.to.1) %in% rownames(ortho.df))
all(rownames(names_1.to.1) %in% ortho.df$orthogroup)
# any()
```

rownames(names_1.to.1)
   [1] "N0.HOG0000000" "N0.HOG0000014" "N0.HOG0000023" "N0.HOG0000026" "N0.HOG0000044"

rownames(ortho.df)
   [1] "N0.HOG0000000" "N0.HOG0000008" "N0.HOG0000009" "N0.HOG0000012" "N0.HOG0000013"
   [6] "N0.HOG0000014" "N0.HOG0000019" "N0.HOG0000020" "N0.HOG0000021" "N0.HOG0000022"
  [11] "N0.HOG0000023" "N0.HOG0000024" "N0.HOG0000026" "N0.HOG0000028" "N0.HOG0000029"


So all of the 1:1 orthogroups are in ortho.df
```{r}
all(names_1.to.1 %in% rownames(ortho.df))
```

```{r}
length(match(names_1.to.1, rownames(ortho.df)))
#length(names_1.to.1 %in% rownames(ortho.df))
#length(names_1.to.1 %in% ortho.df$orthogroup)
```


*match() requires vectors (not dataframes)*
check class of object
class(names_1.to.1)

```{r}
class(names_1.to.1)
#typeof(names_1.to.1)
```

A %in% B
intersect(A,B)
setdiff(A,B)

```{r}
head(rownames(ortho.df) %in% names_1.to.1)
```


```{r}
names_multi.ortho <- setdiff(rownames(ortho.df), names_1.to.1)
str(names_multi.ortho)
```

### multi-orthogroups dataframe
```{r}
multi.ortho  <- orthogroups[names_multi.ortho, ]
head(multi.ortho)
```

Split out contigs from each orthogroup
```{r}
multi.ortho.Sid <- multi.ortho[ , 1:2]

# be aware of the blank space " " after the comma after the split function !!
s <- strsplit(as.character(multi.ortho.Sid$gene_Sid),", ")
head(s, 2)
```

## Siderastrea multi-orthogroup dataframe
```{r}
multi.ortho.Sid <- multi.ortho[ , 1:2]

# be aware of the blank space " " after the comma after the split function !!
multi.ortho.Sid <- separate_rows(multi.ortho.Sid, gene_Sid, sep = ", ")

multi.ortho.Sid[sapply(multi.ortho.Sid, is.character)] <- lapply(multi.ortho.Sid[sapply(multi.ortho.Sid, is.character)], 
                                       as.factor)

multi.ortho.Sid <- multi.ortho.Sid %>% 
  dplyr::rename(gene = gene_Sid)

dim(multi.ortho.Sid)
```

*Multiple genes per orthogroup*
```{r}
head(multi.ortho.Sid)
```


## Porites astreoides multi-orthogroup dataframe
```{r}
multi.ortho.Past <- multi.ortho[ , c(1,3)]
multi.ortho.Past <- separate_rows(multi.ortho.Past, isotig, sep = ", ")

multi.ortho.Past[sapply(multi.ortho.Past, is.character)] <- lapply(multi.ortho.Past[sapply(multi.ortho.Past, is.character)], 
                                       as.factor)

dim(multi.ortho.Past)
```

*Multiple genes per orthogroup*
```{r}
head(multi.ortho.Past)
```


**!!! isotig versus isogroup in P. astreoides transcriptome reference !!!**


*Use inner_join() if you only want to retain observations when they match across both data frames.*

```{r}
multi.ortho.Past.isotigs <- past_seq2iso %>% 
  inner_join(multi.ortho.Past, by = "isotig")

head(multi.ortho.Past.isotigs)
```


### Porites porites multi-orthogroup dataframe
```{r}
multi.ortho.Ppor.isotigs <- seq2iso %>% 
  inner_join(multi.ortho.Past.isotigs, by = "isotig")

multi.ortho.Ppor.isotigs <- multi.ortho.Ppor.isotigs[ , c(3:4,6)]

multi.ortho.Ppor.isotigs <- multi.ortho.Ppor.isotigs %>% 
  dplyr::rename(isotig = isotig_Ppor,
                gene = gene_Ppor)

head(multi.ortho.Ppor.isotigs)
```

Merge Past multi.ortho and Sid multi.ortho
```{r}
library(dplyr)

multi.ortho.Sid.Past.isotigs <- multi.ortho.Sid %>% 
  inner_join(multi.ortho.Past.isotigs, by = "orthogroup")

multi.ortho.Sid.Past.isotigs <- dplyr::rename(multi.ortho.Sid.Past.isotigs, gene.Sid = gene.x, gene.Past = gene.y, isotig.Past = isotig)

head(multi.ortho.Sid.Past.isotigs)

#write.csv(file = "Orthogroups_Sid-Past_multi-ortho_Past-isotigs.csv", x = multi.ortho.Sid.Past.isotigs)
```


##############################################################################

# 2. Merge DEG results table for each contrast


### Ojo-Ojo vs. Ojo-Control
OO_OC

P. astreoides vs. Siderastrea

```{r}
OO_OC_Past <- read.csv("~/Documents/Rprojects/postdoc\ Rprojects/ODU_postdoc_Rprojects/Paper_Ojo_gene-expression/Porites_astreoides/host/Past_Host_DE_genes_OO_OC_padj.1.csv", header = TRUE) 

OO_OC_Past$gene <- as.factor(OO_OC_Past$gene)
head(OO_OC_Past)
```

```{r}
OO_OC_Sid <- read.csv("~/Documents/Rprojects/postdoc\ Rprojects/ODU_postdoc_Rprojects/Paper_Ojo_gene-expression/Siderastrea/host/Sid_Host_DE_genes_OO_OC_padj.1.csv", header = TRUE) 

OO_OC_Sid$gene <- as.factor(OO_OC_Sid$gene)
head(OO_OC_Sid)
```


**Sid**
*Only one DEG in this comparison*

check if any of the values contained in vector A are also in vector B:
```{r}
any(OO_OC_Sid$gene %in% multi.ortho.Sid$gene)
```

Multi-orthogroup
```{r}
length(intersect(OO_OC_Sid$gene, multi.ortho.Sid$gene))
```

1:1 orthogroup
```{r}
length(intersect(OO_OC_Sid$gene, ortho_1.to.1$gene))
```

*Indeed, this gene is in the unassigned gene list, Orthogroups_UnassignedGenes.tsv*


**Past**

Multi-orthogroup
```{r}
length(intersect(OO_OC_Past$gene, multi.ortho.Past.isotigs$gene))
```

1:1 orthogroup
```{r}
length(intersect(OO_OC_Past$gene, ortho.1.to.1_Past.isotigs$gene))
```


```{r}
past_multi.isogroups_OO_OC <- multi.ortho.Past.isotigs %>% 
  inner_join(OO_OC_Past, by = "gene")

dim(past_multi.isogroups_OO_OC)
```

```{r}
head(past_multi.isogroups_OO_OC)
```

**orthogroups that are counted more than once in the DEG results table**
```{r}
past_multi.isogroups_OO_OC %>% dplyr::count(orthogroup) %>% filter(n > 1)
```

Breakdown of which orthogroups are duplicated 
```{r}
past_multi.isogroups_OO_OC %>% 
  group_by(orthogroup) %>% 
  filter(n()>1)
```



1:1 orthogroups
```{r}
past_1.to.1.isogroups_OO_OC <- ortho.1.to.1_Past.isotigs %>% 
  inner_join(OO_OC_Past, by = "gene")

dim(past_1.to.1.isogroups_OO_OC)
```

```{r}
head(past_1.to.1.isogroups_OO_OC)
```

**orthogroups that are counted more than once in the DEG results table**
```{r}
past_1.to.1.isogroups_OO_OC %>% dplyr::count(orthogroup) %>% filter(n > 1)
```

**Result:**
No shared response.


### Make contrast-specific orthogroup dataframe
```{r}
multi.isogroups_OO_OC <- past_multi.isogroups_OO_OC[, 1:3]

multi.isogroups_OO_OC <- multi.isogroups_OO_OC %>% 
  inner_join(multi.ortho.Sid, by = "orthogroup") %>% 
  dplyr::rename(gene.Past = gene.x, gene.Sid = gene.y)

head(multi.isogroups_OO_OC)
```

```{r}
orthogroups.OO_OC <- past_1.to.1.isogroups_OO_OC[, 1:4]

orthogroups.OO_OC <- orthogroups.OO_OC %>% 
  dplyr::rename(gene.Past = gene, gene.Sid = gene_Sid)

head(orthogroups.OO_OC)
```

### merge multi orthogroup with 1:1 orthogroup

**full join**
    - return all rows and columns from the left and right data set
    - unmatched rows from either left data sets receive NA values in the associated new columns

```{r}
orthogroups.OO_OC <- full_join(orthogroups.OO_OC, multi.isogroups_OO_OC, by = c("orthogroup", "gene.Past", "gene.Sid", "isotig"))

dim(orthogroups.OO_OC)
```

```{r}
head(orthogroups.OO_OC)
```

Past genes (same as above)
```{r}
length(intersect(multi.ortho.Past.isotigs$gene, orthogroups.OO_OC$gene.Past))
```

```{r}
length(intersect(multi.ortho.Sid$orthogroup, orthogroups.OO_OC$orthogroup))
```

86 + 35 = 121


```{r}
length(intersect(past_multi.isogroups_OO_OC$orthogroup, orthogroups.OO_OC$orthogroup))
```


## Principal Component Analysis (PCA)
### Pairwise group contrasts

Significantly different genes (pvalue < 0.05) between treatment groups for pairwise comparisons


## Siderastrea

### PCA - All indiv. - OO_OC
```{r}
sigChanges <- orthogroups.OO_OC$gene.Sid
counts_de <- Sid.counts.transformed.vst[rownames(Sid.counts.transformed.vst) %in% sigChanges,]
dim(counts_de)
```

```{r}
# save plot
#pdf("Sid_Host_PCA_all-indiv_orthogroups_OO_OC.pdf")

pcadata = DESeq2::plotPCA(counts_de, intgroup = c("group"), returnData = TRUE)
percentVar = round(100 * attr(pcadata, "percentVar"))
pca = prcomp(t(assay(counts_de)), center = TRUE, scale. = FALSE)

plot <- DESeq2::plotPCA(counts_de, returnData = TRUE, intgroup = c("group")) %>% 
      ggplot(aes(x = PC1, y = PC2)) +
      geom_point(aes(color = group), size = 1.3) +
      stat_ellipse(geom = "polygon", alpha = 1/10, aes(fill = group)) +
      scale_color_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) + #chosen.palette
      scale_fill_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) +
      xlab(paste0("PC1: ",percentVar[1],"% variance")) +
      ylab(paste0("PC2: ",percentVar[2],"% variance")) +
      labs(title = "Ojo-Ojo vs. Ojo-Control (orthogroups)") +
      labs(color = "Origin.Destination", fill = "Origin.Destination") +
      theme_cowplot()

plot
```


### PCA - Subset indiv. - OO_OC
```{r}
# select relevant sample columns for the given pairwise comparison
colnames(Sid.counts.transformed.vst)
counts_de <- Sid.counts.transformed.vst[,c(12:22)]
counts_de <- counts_de[rownames(counts_de) %in% sigChanges,]
dim(counts_de)
```


```{r}
# save plot
#pdf("Sid_Host_PCA_subset_orthogroups_OO_OC.pdf")

pcadata = DESeq2::plotPCA(counts_de, intgroup = c("group"), returnData = TRUE)
percentVar = round(100 * attr(pcadata, "percentVar"))
pca = prcomp(t(assay(counts_de)), center = TRUE, scale. = FALSE)

plot <- DESeq2::plotPCA(counts_de, returnData = TRUE, intgroup = c("group")) %>% 
      ggplot(aes(x = PC1, y = PC2)) +
      geom_point(aes(color = group), size = 1.3) +
      stat_ellipse(geom = "polygon", alpha = 1/10, aes(fill = group)) +
      scale_color_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) + #chosen.palette
      scale_fill_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) +
      xlab(paste0("PC1: ",percentVar[1],"% variance")) +
      ylab(paste0("PC2: ",percentVar[2],"% variance")) +
      labs(title = "Ojo-Ojo vs. Ojo-Control (orthogroups)") +
      labs(color = "Origin.Destination", fill = "Origin.Destination") +
      theme_cowplot()

plot

#dev.off()
```

check
```{r}
dim(counts_de)
colnames(counts_de)
```


## Porites astreoides

### PCA - All indiv. - OO_OC
```{r}
sigChanges <- orthogroups.OO_OC$gene.Past
counts_de <- Past.counts.transformed.vst[rownames(Past.counts.transformed.vst) %in% sigChanges,]
dim(counts_de)
```

```{r}
# save plot
#pdf("Past_Host_PCA_all-indiv_orthogroups_OO_OC.pdf")

pcadata = DESeq2::plotPCA(counts_de, intgroup = c("group"), returnData = TRUE)
percentVar = round(100 * attr(pcadata, "percentVar"))
pca = prcomp(t(assay(counts_de)), center = TRUE, scale. = FALSE)

plot <- DESeq2::plotPCA(counts_de, returnData = TRUE, intgroup = c("group")) %>% 
      ggplot(aes(x = PC1, y = PC2)) +
      geom_point(aes(color = group), size = 1.3) +
      stat_ellipse(geom = "polygon", alpha = 1/10, aes(fill = group)) +
      scale_color_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) + #chosen.palette
      scale_fill_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) +
      xlab(paste0("PC1: ",percentVar[1],"% variance")) +
      ylab(paste0("PC2: ",percentVar[2],"% variance")) +
      labs(title = "Ojo-Ojo vs. Ojo-Control (orthogroups)") +
      labs(color = "Origin.Destination", fill = "Origin.Destination") +
      theme_cowplot()

plot
```


### PCA - Subset indiv. - OO_OC
```{r}
# select relevant sample columns for the given pairwise comparison
colnames(Past.counts.transformed.vst)
counts_de <- Past.counts.transformed.vst[,c(10:16,22:29)]
counts_de <- counts_de[rownames(counts_de) %in% sigChanges,]
dim(counts_de)
```


```{r}
# save plot
#pdf("Past_Host_PCA_subset_orthogroups_OO_OC.pdf")

pcadata = DESeq2::plotPCA(counts_de, intgroup = c("group"), returnData = TRUE)
percentVar = round(100 * attr(pcadata, "percentVar"))
pca = prcomp(t(assay(counts_de)), center = TRUE, scale. = FALSE)

plot <- DESeq2::plotPCA(counts_de, returnData = TRUE, intgroup = c("group")) %>% 
      ggplot(aes(x = PC1, y = PC2)) +
      geom_point(aes(color = group), size = 1.3) +
      stat_ellipse(geom = "polygon", alpha = 1/10, aes(fill = group)) +
      scale_color_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) + #chosen.palette
      scale_fill_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) +
      xlab(paste0("PC1: ",percentVar[1],"% variance")) +
      ylab(paste0("PC2: ",percentVar[2],"% variance")) +
      labs(title = "Ojo-Ojo vs. Ojo-Control (orthogroups)") +
      labs(color = "Origin.Destination", fill = "Origin.Destination") +
      theme_cowplot()

plot

#dev.off()
```

check
```{r}
dim(counts_de)
colnames(counts_de)
```



##############################################################################

### Ojo-Ojo vs. Lagoon-Ojo
OO_LO

P. astreoides vs. Siderastrea

```{r}
OO_LO_Past <- read.csv("~/Documents/Rprojects/postdoc\ Rprojects/ODU_postdoc_Rprojects/Paper_Ojo_gene-expression/Porites_astreoides/host/Past_Host_DE_genes_OO_LO_padj.1.csv", header = TRUE) 

OO_LO_Past$gene <- as.factor(OO_LO_Past$gene)
head(OO_LO_Past)
```

```{r}
OO_LO_Sid <- read.csv("~/Documents/Rprojects/postdoc\ Rprojects/ODU_postdoc_Rprojects/Paper_Ojo_gene-expression/Siderastrea/host/Sid_Host_DE_genes_OO_LO_padj.1.csv", header = TRUE) 

OO_LO_Sid$gene <- as.factor(OO_LO_Sid$gene)
head(OO_LO_Sid)
```


**Sid**
*Only two DEG in this comparison*

check if any of the values contained in vector A are also in vector B:
```{r}
any(OO_LO_Sid$gene %in% multi.ortho.Sid$gene)
```

Multi-orthogroup
```{r}
length(intersect(OO_LO_Sid$gene, multi.ortho.Sid$gene))
```

1:1 orthogroup
```{r}
length(intersect(OO_LO_Sid$gene, ortho_1.to.1$gene))
```

*Indeed, both these genes are in the unassigned gene list, Orthogroups_UnassignedGenes.tsv*


**Past**

Multi-orthogroup
```{r}
length(intersect(OO_LO_Past$gene, multi.ortho.Past.isotigs$gene))
```

1:1 orthogroup
```{r}
length(intersect(OO_LO_Past$gene, ortho.1.to.1_Past.isotigs$gene))
```

```{r}
past_multi.isogroups_OO_LO <- multi.ortho.Past.isotigs %>% 
  inner_join(OO_LO_Past, by = "gene")

dim(past_multi.isogroups_OO_LO)
```

```{r}
head(past_multi.isogroups_OO_LO)
```

**orthogroups that are counted more than once in the DEG results table**
```{r}
past_multi.isogroups_OO_LO %>% dplyr::count(orthogroup) %>% filter(n > 1)
```

Breakdown of which orthogroups are duplicated 
```{r}
past_multi.isogroups_OO_LO %>% 
  group_by(orthogroup) %>% 
  filter(n()>1)
```


1:1 Past
```{r}
past_1.to.1.isogroups_OO_LO <- ortho.1.to.1_Past.isotigs %>% 
  inner_join(OO_LO_Past, by = "gene")

dim(past_1.to.1.isogroups_OO_LO)
```

```{r}
head(past_1.to.1.isogroups_OO_LO)
```

**orthogroups that are counted more than once in the DEG results table**
```{r}
past_1.to.1.isogroups_OO_LO %>% dplyr::count(orthogroup) %>% filter(n > 1)
```

**Result:**
No shared response.


### Make contrast-specific orthogroup dataframe
```{r}
multi.isogroups_OO_LO <- past_multi.isogroups_OO_LO[, 1:3]

multi.isogroups_OO_LO <- multi.isogroups_OO_LO %>% 
  inner_join(multi.ortho.Sid, by = "orthogroup") %>% 
  dplyr::rename(gene.Past = gene.x, gene.Sid = gene.y)

head(multi.isogroups_OO_LO)
```

```{r}
orthogroups.OO_LO <- past_1.to.1.isogroups_OO_LO[, 1:4]

orthogroups.OO_LO <- orthogroups.OO_LO %>% 
  dplyr::rename(gene.Past = gene, gene.Sid = gene_Sid)

head(orthogroups.OO_LO)
```

### merge multi orthogroup with 1:1 orthogroup

**full join**
    - return all rows and columns from the left and right data set
    - unmatched rows from either left data sets receive NA values in the associated new columns

```{r}
orthogroups.OO_LO <- full_join(orthogroups.OO_LO, multi.isogroups_OO_LO, by = c("orthogroup", "gene.Past", "gene.Sid", "isotig"))

dim(orthogroups.OO_LO)
```

```{r}
head(orthogroups.OO_LO)
```

Past genes (same as above)
```{r}
length(intersect(multi.ortho.Past.isotigs$gene, orthogroups.OO_LO$gene.Past))
```

```{r}
length(intersect(multi.ortho.Sid$orthogroup, orthogroups.OO_LO$orthogroup))
```


```{r}
length(intersect(past_multi.isogroups_OO_LO$orthogroup, orthogroups.OO_LO$orthogroup))
```


## Principal Component Analysis (PCA)
### Pairwise group contrasts

Significantly different genes (pvalue < 0.05) between treatment groups for pairwise comparisons


## Siderastrea

### PCA - All indiv. - OO_LO
```{r}
sigChanges <- orthogroups.OO_LO$gene.Sid
counts_de <- Sid.counts.transformed.vst[rownames(Sid.counts.transformed.vst) %in% sigChanges,]
dim(counts_de)
```

```{r}
# save plot
#pdf("Sid_Host_PCA_all-indiv_orthogroups_OO_LO.pdf")

pcadata = DESeq2::plotPCA(counts_de, intgroup = c("group"), returnData = TRUE)
percentVar = round(100 * attr(pcadata, "percentVar"))
pca = prcomp(t(assay(counts_de)), center = TRUE, scale. = FALSE)

plot <- DESeq2::plotPCA(counts_de, returnData = TRUE, intgroup = c("group")) %>% 
      ggplot(aes(x = PC1, y = PC2)) +
      geom_point(aes(color = group), size = 1.3) +
      stat_ellipse(geom = "polygon", alpha = 1/10, aes(fill = group)) +
      scale_color_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) + #chosen.palette
      scale_fill_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) +
      xlab(paste0("PC1: ",percentVar[1],"% variance")) +
      ylab(paste0("PC2: ",percentVar[2],"% variance")) +
      labs(title = "Ojo-Ojo vs. Lagoon-Ojo (orthogroups)") +
      labs(color = "Origin.Destination", fill = "Origin.Destination") +
      theme_cowplot()

plot
```


### PCA - Subset indiv. - OO_LO
```{r}
# select relevant sample columns for the given pairwise comparison
colnames(Sid.counts.transformed.vst)
counts_de <- Sid.counts.transformed.vst[,c(2,4,6,9,13,15,17,19,21,22)]
counts_de <- counts_de[rownames(counts_de) %in% sigChanges,]
dim(counts_de)
```


```{r}
# save plot
#pdf("Sid_Host_PCA_subset_orthogroups_OO_LO.pdf")

pcadata = DESeq2::plotPCA(counts_de, intgroup = c("group"), returnData = TRUE)
percentVar = round(100 * attr(pcadata, "percentVar"))
pca = prcomp(t(assay(counts_de)), center = TRUE, scale. = FALSE)

plot <- DESeq2::plotPCA(counts_de, returnData = TRUE, intgroup = c("group")) %>% 
      ggplot(aes(x = PC1, y = PC2)) +
      geom_point(aes(color = group), size = 1.3) +
      stat_ellipse(geom = "polygon", alpha = 1/10, aes(fill = group)) +
      scale_color_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) + #chosen.palette
      scale_fill_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) +
      xlab(paste0("PC1: ",percentVar[1],"% variance")) +
      ylab(paste0("PC2: ",percentVar[2],"% variance")) +
      labs(title = "Ojo-Ojo vs. Lagoon-Ojo (orthogroups)") +
      labs(color = "Origin.Destination", fill = "Origin.Destination") +
      theme_cowplot()

plot

#dev.off()
```

check
```{r}
dim(counts_de)
colnames(counts_de)
```

## Porites astreoides

### PCA - All indiv. - OO_LO
```{r}
sigChanges <- orthogroups.OO_LO$gene.Past
counts_de <- Past.counts.transformed.vst[rownames(Past.counts.transformed.vst) %in% sigChanges,]
dim(counts_de)
```

```{r}
# save plot
#pdf("Past_Host_PCA_all-indiv_orthogroups_OO_LO.pdf")

pcadata = DESeq2::plotPCA(counts_de, intgroup = c("group"), returnData = TRUE)
percentVar = round(100 * attr(pcadata, "percentVar"))
pca = prcomp(t(assay(counts_de)), center = TRUE, scale. = FALSE)

plot <- DESeq2::plotPCA(counts_de, returnData = TRUE, intgroup = c("group")) %>% 
      ggplot(aes(x = PC1, y = PC2)) +
      geom_point(aes(color = group), size = 1.3) +
      stat_ellipse(geom = "polygon", alpha = 1/10, aes(fill = group)) +
      scale_color_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) + #chosen.palette
      scale_fill_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) +
      xlab(paste0("PC1: ",percentVar[1],"% variance")) +
      ylab(paste0("PC2: ",percentVar[2],"% variance")) +
      labs(title = "Ojo-Ojo vs. Lagoon-Ojo (orthogroups)") +
      labs(color = "Origin.Destination", fill = "Origin.Destination") +
      theme_cowplot()

plot
```


### PCA - Subset indiv. - OO_LO
```{r}
# select relevant sample columns for the given pairwise comparison
colnames(Past.counts.transformed.vst)
counts_de <- Past.counts.transformed.vst[,c(2,4,6,9,13,15,17,19,21,22)]
counts_de <- counts_de[rownames(counts_de) %in% sigChanges,]
dim(counts_de)
```


```{r}
# save plot
#pdf("Past_Host_PCA_subset_orthogroups_OO_LO.pdf")

pcadata = DESeq2::plotPCA(counts_de, intgroup = c("group"), returnData = TRUE)
percentVar = round(100 * attr(pcadata, "percentVar"))
pca = prcomp(t(assay(counts_de)), center = TRUE, scale. = FALSE)

plot <- DESeq2::plotPCA(counts_de, returnData = TRUE, intgroup = c("group")) %>% 
      ggplot(aes(x = PC1, y = PC2)) +
      geom_point(aes(color = group), size = 1.3) +
      stat_ellipse(geom = "polygon", alpha = 1/10, aes(fill = group)) +
      scale_color_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) + #chosen.palette
      scale_fill_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) +
      xlab(paste0("PC1: ",percentVar[1],"% variance")) +
      ylab(paste0("PC2: ",percentVar[2],"% variance")) +
      labs(title = "Ojo-Ojo vs. Lagoon-Ojo (orthogroups)") +
      labs(color = "Origin.Destination", fill = "Origin.Destination") +
      theme_cowplot()

plot

#dev.off()
```

check
```{r}
dim(counts_de)
colnames(counts_de)
```



##############################################################################

### Lagoon-Ojo vs. Lagoon-Control
LO_LC

P. astreoides vs. Siderastrea

```{r}
LO_LC_Past <- read.csv("~/Documents/Rprojects/postdoc\ Rprojects/ODU_postdoc_Rprojects/Paper_Ojo_gene-expression/Porites_astreoides/host/Past_Host_DE_genes_LO_LC_padj.1.csv", header = TRUE) 

LO_LC_Past$gene <- as.factor(LO_LC_Past$gene)
head(LO_LC_Past)
```

*This contrast has zero differentially expressed genes for Siderastrea*
```{r}
# LO_LC_Sid <- read.csv("~/Documents/Rprojects/postdoc\ Rprojects/ODU_postdoc_Rprojects/Paper_Ojo_gene-expression/Siderastrea/host/Sid_Host_DE_genes_LO_LC_padj.1.csv", header = TRUE) 
# 
# LO_LC_Sid$gene <- as.factor(LO_LC_Sid$gene)
# head(LO_LC_Sid)
```

Result:
Cannot compare because Siderastrea had zero DEG for this contrast.



##############################################################################

### Ojo-Control vs. Lagoon-Control
OC_LC

P. astreoides vs. Siderastrea

```{r}
OC_LC_Past <- read.csv("~/Documents/Rprojects/postdoc\ Rprojects/ODU_postdoc_Rprojects/Paper_Ojo_gene-expression/Porites_astreoides/host/Past_Host_DE_genes_OC_LC_padj.1.csv", header = TRUE) 

OC_LC_Past$gene <- as.factor(OC_LC_Past$gene)
head(OC_LC_Past)
```

```{r}
OC_LC_Sid <- read.csv("~/Documents/Rprojects/postdoc\ Rprojects/ODU_postdoc_Rprojects/Paper_Ojo_gene-expression/Siderastrea/host/Sid_Host_DE_genes_OC_LC_padj.1.csv", header = TRUE) 

OC_LC_Sid$gene <- as.factor(OC_LC_Sid$gene)
head(OC_LC_Sid)
```


**Sid**
*Only one DEG in this comparison*

check if any of the values contained in vector A are also in vector B:
```{r}
any(OC_LC_Sid$gene %in% multi.ortho.Sid$gene)
```

Multi-orthogroup list
```{r}
length(intersect(OC_LC_Sid$gene, multi.ortho.Sid$gene))
```

1:1 orthogroup
```{r}
length(intersect(OC_LC_Sid$gene, ortho_1.to.1$gene))
```

*Indeed, this gene is in the unassigned gene list, Orthogroups_UnassignedGenes.tsv*


**Past**
*Only one DEG in this comparison*

Multi-orthogroup list
```{r}
length(intersect(OC_LC_Past$gene, multi.ortho.Past.isotigs$gene))
```

1:1 orthogroup
```{r}
length(intersect(OC_LC_Past$gene, ortho.1.to.1_Past.isotigs$gene))
```


```{r}
past_1.to.1.isogroups_OC_LC <- ortho.1.to.1_Past.isotigs %>% 
  inner_join(OC_LC_Past, by = "gene")

dim(past_1.to.1.isogroups_OC_LC)
```

```{r}
head(past_1.to.1.isogroups_OC_LC)
```


**Result:**
No shared response.


Did not do a PCA because only 1 orthogroup per species.


##############################################################################

### Reef-Ojo vs. Reef-Control
RO_RC

P. porites vs. Siderastrea

```{r}
RO_RC_Ppor <- read.csv("~/Documents/Rprojects/postdoc\ Rprojects/ODU_postdoc_Rprojects/Paper_Ojo_gene-expression/Porites_porites/host/Ppor_Host_DE_genes_RO_RC_padj.1.csv", header = TRUE) 

RO_RC_Ppor$gene <- as.factor(RO_RC_Ppor$gene)
head(RO_RC_Ppor)
```

```{r}
RO_RC_Sid <- read.csv("~/Documents/Rprojects/postdoc\ Rprojects/ODU_postdoc_Rprojects/Paper_Ojo_gene-expression/Siderastrea/host/Sid_Host_DE_genes_RO_RC_padj.1.csv", header = TRUE) 

RO_RC_Sid$gene <- as.factor(RO_RC_Sid$gene)
head(RO_RC_Sid)
```


**Sid**

check if any of the values contained in vector A are also in vector B:
```{r}
any(RO_RC_Sid$gene %in% multi.ortho.Sid$gene)
```

Multi-orthogroup list
```{r}
length(intersect(RO_RC_Sid$gene, multi.ortho.Sid$gene))
```

1:1 orthogroup
```{r}
length(intersect(RO_RC_Sid$gene, ortho_1.to.1$gene))
```

```{r}
sid_multi.isogroups_RO_RC <- multi.ortho.Sid %>% 
  inner_join(RO_RC_Sid, by = "gene")

dim(sid_multi.isogroups_RO_RC)
```

```{r}
head(sid_multi.isogroups_RO_RC)
```

**orthogroups that are counted more than once in the DEG results table**
```{r}
sid_multi.isogroups_RO_RC %>% dplyr::count(orthogroup) %>% filter(n > 1)
```

Breakdown of which orthogroups are duplicated 
```{r}
sid_multi.isogroups_RO_RC %>% 
  group_by(orthogroup) %>% 
  filter(n()>1)
```


```{r}
ortho_1.to.1 <- ortho_1.to.1 %>% 
  dplyr::rename(gene = gene_Sid)
```



1:1 orthogroups
```{r}
sid_1.to.1.isogroups_RO_RC <- ortho_1.to.1 %>% 
  inner_join(RO_RC_Sid, by = "gene")

dim(sid_1.to.1.isogroups_RO_RC)
```

```{r}
head(sid_1.to.1.isogroups_RO_RC)
```

**orthogroups that are counted more than once in the DEG results table**
```{r}
sid_1.to.1.isogroups_RO_RC %>% dplyr::count(orthogroup) %>% filter(n > 1)
```



**P. porites**

Multi-orthogroup list
```{r}
length(intersect(RO_RC_Ppor$gene, multi.ortho.Ppor.isotigs$gene))
```


```{r}
ppor_multi.isogroups_RO_RC <- multi.ortho.Ppor.isotigs %>% 
  inner_join(RO_RC_Ppor, by = "gene")

dim(ppor_multi.isogroups_RO_RC)
```

```{r}
head(ppor_multi.isogroups_RO_RC)
```

**orthogroups that are counted more than once in the DEG results table**
```{r}
ppor_multi.isogroups_RO_RC %>% dplyr::count(orthogroup) %>% filter(n > 1)
```

Breakdown of which orthogroups are duplicated 
```{r}
ppor_multi.isogroups_RO_RC %>% 
  group_by(orthogroup) %>% 
  filter(n()>1)
```



1:1 orthogroup
```{r}
length(intersect(RO_RC_Ppor$gene, ortho.1.to.1_Ppor.isotigs$gene))
```


Check for overlapping orthogroups between multi-orthogroup Sid and P.por
```{r}
length(intersect(sid_multi.isogroups_RO_RC$orthogroup, ppor_multi.isogroups_RO_RC$orthogroup))
```


**Result:**
No shared DEG.


### Make contrast-specific orthogroup dataframe

```{r}
ppor_multi.isogroups_RO_RC <- ppor_multi.isogroups_RO_RC[, 1:3]

ppor_multi.isogroups_RO_RC <- ppor_multi.isogroups_RO_RC %>% 
  inner_join(multi.ortho.Sid, by = "orthogroup") %>% 
  dplyr::rename(gene.Ppor = gene.x, gene.Sid = gene.y)

head(ppor_multi.isogroups_RO_RC)
```


### merge multi orthogroup with 1:1 orthogroup

```{r}
multi.isogroups_RO_RC.sid <- sid_multi.isogroups_RO_RC[, 1:2]

multi.isogroups_RO_RC.sid <- multi.isogroups_RO_RC.sid %>%
  dplyr::rename(gene.Sid = gene)

orthogroups.RO_RC <- full_join(multi.isogroups_RO_RC.sid, ppor_multi.isogroups_RO_RC, by = c( "orthogroup", "gene.Sid"))

head(orthogroups.RO_RC)
```

1.to.1.isogroups
```{r}
orthogroups_1.to.1_RO_RC <- sid_1.to.1.isogroups_RO_RC[, 1:2]

orthogroups_1.to.1_RO_RC <- orthogroups_1.to.1_RO_RC %>% 
  dplyr::rename(gene.Sid = gene)

head(orthogroups_1.to.1_RO_RC)
```


```{r}
orthogroups.RO_RC <- full_join(orthogroups.RO_RC, orthogroups_1.to.1_RO_RC, by = c( "orthogroup", "gene.Sid"))
head(orthogroups.RO_RC)
```




Ppor genes (same as above)
```{r}
length(intersect(multi.ortho.Ppor.isotigs$gene, orthogroups.RO_RC$gene.Ppor))
```


```{r}
length(intersect(multi.isogroups_RO_RC.sid$orthogroup, orthogroups.RO_RC$orthogroup))
```


## Principal Component Analysis (PCA)
### Pairwise group contrasts

Significantly different genes (pvalue < 0.05) between treatment groups for pairwise comparisons


## Siderastrea

### PCA - All indiv. - RO_RC
```{r}
sigChanges <- orthogroups.RO_RC$gene.Sid
counts_de <- Sid.counts.transformed.vst[rownames(Sid.counts.transformed.vst) %in% sigChanges,]
dim(counts_de)
```

```{r}
# save plot
#pdf("Sid_Host_PCA_all-indiv_orthogroups_RO_RC.pdf")

pcadata = DESeq2::plotPCA(counts_de, intgroup = c("group"), returnData = TRUE)
percentVar = round(100 * attr(pcadata, "percentVar"))
pca = prcomp(t(assay(counts_de)), center = TRUE, scale. = FALSE)

plot <- DESeq2::plotPCA(counts_de, returnData = TRUE, intgroup = c("group")) %>% 
      ggplot(aes(x = PC1, y = PC2)) +
      geom_point(aes(color = group), size = 1.3) +
      stat_ellipse(geom = "polygon", alpha = 1/10, aes(fill = group)) +
      scale_color_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) + #chosen.palette
      scale_fill_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) +
      xlab(paste0("PC1: ",percentVar[1],"% variance")) +
      ylab(paste0("PC2: ",percentVar[2],"% variance")) +
      labs(title = "Reef-Ojo vs. Reef-Control (orthogroups)") +
      labs(color = "Origin.Destination", fill = "Origin.Destination") +
      theme_cowplot()

plot
```


### PCA - Subset indiv. - RO_RC
```{r}
# select relevant sample columns for the given pairwise comparison
colnames(Sid.counts.transformed.vst)
counts_de <- Sid.counts.transformed.vst[,c(23:29)]
counts_de <- counts_de[rownames(counts_de) %in% sigChanges,]
dim(counts_de)
```


```{r}
# save plot
#pdf("Sid_Host_PCA_subset_orthogroups_RO_RC.pdf")

pcadata = DESeq2::plotPCA(counts_de, intgroup = c("group"), returnData = TRUE)
percentVar = round(100 * attr(pcadata, "percentVar"))
pca = prcomp(t(assay(counts_de)), center = TRUE, scale. = FALSE)

plot <- DESeq2::plotPCA(counts_de, returnData = TRUE, intgroup = c("group")) %>% 
      ggplot(aes(x = PC1, y = PC2)) +
      geom_point(aes(color = group), size = 1.3) +
      stat_ellipse(geom = "polygon", alpha = 1/10, aes(fill = group)) +
      scale_color_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) + #chosen.palette
      scale_fill_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) +
      xlab(paste0("PC1: ",percentVar[1],"% variance")) +
      ylab(paste0("PC2: ",percentVar[2],"% variance")) +
      labs(title = "Reef-Ojo vs. Reef-Control (orthogroups)") +
      labs(color = "Origin.Destination", fill = "Origin.Destination") +
      theme_cowplot()

plot

#dev.off()
```

check
```{r}
dim(counts_de)
colnames(counts_de)
```

## Porites porites

### PCA - All indiv. - RO_RC
```{r}
sigChanges <- orthogroups.RO_RC$gene.Ppor
counts_de <- Ppor.counts.transformed.vst[rownames(Ppor.counts.transformed.vst) %in% sigChanges,]
dim(counts_de)
```

```{r}
# save plot
#pdf("Ppor_Host_PCA_all-indiv_orthogroups_RO_RC.pdf")

pcadata = DESeq2::plotPCA(counts_de, intgroup = c("group"), returnData = TRUE)
percentVar = round(100 * attr(pcadata, "percentVar"))
pca = prcomp(t(assay(counts_de)), center = TRUE, scale. = FALSE)

plot <- DESeq2::plotPCA(counts_de, returnData = TRUE, intgroup = c("group")) %>% 
      ggplot(aes(x = PC1, y = PC2)) +
      geom_point(aes(color = group), size = 1.3) +
      stat_ellipse(geom = "polygon", alpha = 1/10, aes(fill = group)) +
      scale_color_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) + #chosen.palette
      scale_fill_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) +
      xlab(paste0("PC1: ",percentVar[1],"% variance")) +
      ylab(paste0("PC2: ",percentVar[2],"% variance")) +
      labs(title = "Reef-Ojo vs. Reef-Control (orthogroups)") +
      labs(color = "Origin.Destination", fill = "Origin.Destination") +
      theme_cowplot()

plot
```


### PCA - Subset indiv. - OO_LO
```{r}
# select relevant sample columns for the given pairwise comparison
colnames(Ppor.counts.transformed.vst)
counts_de <- counts_de[rownames(counts_de) %in% sigChanges,]
dim(counts_de)
```


```{r}
# save plot
#pdf("Ppor_Host_PCA_subset_orthogroups_RO_RC.pdf")

pcadata = DESeq2::plotPCA(counts_de, intgroup = c("group"), returnData = TRUE)
percentVar = round(100 * attr(pcadata, "percentVar"))
pca = prcomp(t(assay(counts_de)), center = TRUE, scale. = FALSE)

plot <- DESeq2::plotPCA(counts_de, returnData = TRUE, intgroup = c("group")) %>% 
      ggplot(aes(x = PC1, y = PC2)) +
      geom_point(aes(color = group), size = 1.3) +
      stat_ellipse(geom = "polygon", alpha = 1/10, aes(fill = group)) +
      scale_color_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) + #chosen.palette
      scale_fill_manual(values = c("dodgerblue", "gold2", "#163343", "firebrick", "forestgreen", "bisque3")) +
      xlab(paste0("PC1: ",percentVar[1],"% variance")) +
      ylab(paste0("PC2: ",percentVar[2],"% variance")) +
      labs(title = "Reef-Ojo vs. Reef-Control (orthogroups)") +
      labs(color = "Origin.Destination", fill = "Origin.Destination") +
      theme_cowplot()

plot

#dev.off()
```

check
```{r}
dim(counts_de)
colnames(counts_de)
```



##############################################################################

If there are any shared DEG:

## filter by DEG direction 


### up-regulated

```{r}
# OO_OC_up <- OO_OC %>%
#     filter(baseMean > 0 & padj < 0.1)
```


### down-regulated

```{r}
# OO_OC_down <- OO_OC %>%
#     filter(baseMean < 0 & padj < 0.1)
```


##############################################################################

# Annotated transcriptomes

```{r}
past.annot <- read.csv("~/Documents/Postdoc/ODU_postdoc/projects/Ojo_gene-expression/Porites_astreoides/Past_hybridreference_totalannotated.csv")

#past.annot[sapply(past.annot, is.character)] <- lapply(past.annot[sapply(past.annot, is.character)], as.factor)
```


Dataframe with only isotigs 
```{r}
past.annot.contigs <- past.annot %>% 
  dplyr::filter(grepl("isotig", ContigName))

#past.annot.contigs <- dplyr::rename(past.annot.contigs, isotig = ContigName)

past.annot.contigs$isotig = past.annot.contigs$ContigName

head(past.annot.contigs)
```

Match isotigs with genes
```{r}
past.annot.contigs <- past_seq2iso %>% 
  inner_join(past.annot.contigs, by = "isotig")

head(past.annot.contigs)
```

Dataframe with non-isotig genes
```{r}
past.annot.genes <- past.annot %>% 
  dplyr::filter(!grepl("isotig", ContigName))

#past.annot.genes <- dplyr::rename(past.annot.genes, gene = ContigName)

past.annot.genes$gene <- past.annot.genes$ContigName
past.annot.genes$isotig <- past.annot.genes$ContigName

dim(past.annot.genes)
```

**full join**
    - return all rows and columns from the left and right data set
    - unmatched rows from either left data sets receive NA values in the associated new columns

```{r}
past.annot.genes.final <- full_join(past.annot.contigs, past.annot.genes, by = c("gene", "ContigLength", "topnrMatch", "topnrEvalue", "topnrAlignmentLength", "topnrPercMatch", "nobadwordnrMatch", "nobadwordnrEvalue", "nobadwordnrAlignLength", "nobadwordnrPercentMatch", "Uniprotmatch", "X._identity", "evalue",  "ID", "Description",  "KEGG", "KEGGKO", "PFAM", "GO", "GOCC", "GOBP", "GOMF", "Keywords"))

dim(past.annot.genes.final)
```

```{r}
#write.csv(file = "Past_hybridreference_totalannotated_isotig-Gene.csv", x = past.annot.genes.final)
```


Isotig - Gene - ContigName
```{r}
# merge back non-isotig genes so we have full gene names (to relate to DEG table)
past.annot.final <- full_join(past.annot.contigs, past.annot.genes, by = c("isotig" ,"gene", "ContigName", "ContigLength", "topnrMatch", "topnrEvalue", "topnrAlignmentLength", "topnrPercMatch", "nobadwordnrMatch", "nobadwordnrEvalue", "nobadwordnrAlignLength", "nobadwordnrPercentMatch", "Uniprotmatch", "X._identity", "evalue",  "ID", "Description",  "KEGG", "KEGGKO", "PFAM", "GO", "GOCC", "GOBP", "GOMF", "Keywords")) 


head(past.annot.final)
```

```{r}
#write.csv(file = "Past_hybridreference_totalannotated_Final.csv", x = past.annot.final)
```


##############################################################################

### Session Info
```{r}
sessionInfo()
```
